pkg/envoy/rds/builder.go

```diff
@@ -0,0 +1,80 @@
+package rds
+
+import (
+	"github.com/envoyproxy/go-control-plane/pkg/cache/types"
+
+	"github.com/openservicemesh/osm/pkg/configurator"
+	"github.com/openservicemesh/osm/pkg/envoy"
+	"github.com/openservicemesh/osm/pkg/envoy/rds/route"
+	"github.com/openservicemesh/osm/pkg/trafficpolicy"
+)
+
+type routesBuilder struct {
+	inboundMeshTrafficPolicy *trafficpolicy.InboundMeshTrafficPolicy
+	outboundMeshTrafficPolicy *trafficpolicy.OutboundMeshTrafficPolicy
+	ingressTrafficPolicies []*trafficpolicy.InboundTrafficPolicy
+	egressTrafficPolicy *trafficpolicy.EgressTrafficPolicy
+	proxy *envoy.Proxy
+	cfg configurator.Configurator
+	trustDomain string
+}
+
+func newRoutesBuilder(inboundMeshTrafficPolicy *trafficpolicy.InboundMeshTrafficPolicy, outboundMeshTrafficPolicy *trafficpolicy.OutboundMeshTrafficPolicy, ingressTrafficPolicies []*trafficpolicy.InboundTrafficPolicy, egressTrafficPolicy *trafficpolicy.EgressTrafficPolicy, proxy *envoy.Proxy, cfg configurator.Configurator, trustDomain string) *routesBuilder {
+	return &routesBuilder {
+		inboundMeshTrafficPolicy: inboundMeshTrafficPolicy,
+		outboundMeshTrafficPolicy: outboundMeshTrafficPolicy,
+		ingressTrafficPolicies: ingressTrafficPolicies,
+		egressTrafficPolicy: egressTrafficPolicy,
+		proxy: proxy,
+		cfg: cfg,
+		trustDomain: trustDomain,
+	}
+}
+
+func (b *routesBuilder) Build() ([]types.Resource, error) {
+
+	var rdsResources []types.Resource
+
+	// ---
+	// Build inbound mesh route configurations. These route configurations allow
+	// the services associated with this proxy to accept traffic from downstream
+	// clients on allowed routes.
+	if b.inboundMeshTrafficPolicy != nil {
+		inboundMeshRouteConfig := route.BuildInboundMeshRouteConfiguration(b.inboundMeshTrafficPolicy.HTTPRouteConfigsPerPort, b.proxy, b.cfg, b.trustDomain)
```

Comment:
instead of passing these to the route package's `BuildInboundMeshRouteConfiguration`, let's make BuildInboundMeshRouteConfiguration an unexported method on the builder object.

From the [design doc](https://docs.google.com/document/d/1L_u-XlVU8Fog-uFS3Uf0O2nz8IArQy82cd6Npxm038Y/edit#heading=h.86n6n6h1rqmo), we call out that it's easier when leveraging code navigation to highlight ie: `b.proxy`, and see all of its uses. Whereas when we pass it to another function, we lose that ability.

Reply:
Makes sense, this is sort of what I was alluding to in my top comment (https://github.com/openservicemesh/osm/pull/4973#issuecomment-1208494343) about merging / borrowing some of the methods in route_config.go and implementing that in builder.go instead. 

Let's discuss more over a call, I think that would be easier.